### ISSUE TITLE
cannot compile vf-gltransition.c

### ISSUE THREAD

**Issue by yongfengnice** [transitive-bullshit/ffmpeg-gl-transition#10]
cannot compile vf-gltransition.c, how can I solve this problem?

**Comment by transitive-bullshit** [comment]
You're going to need to include a lot more details.

What platform? What's the error?

**Make sure you can compile a fresh clone of ffmpeg before trying to compile with ffmpeg-gl-transition additions.**

**Comment by yongfengnice** [comment]
![2018-03-15_112104](https://user-images.githubusercontent.com/8778433/37442789-3a265ae0-2843-11e8-98e3-b2bebe6e7792.png)
![2018-03-15_112152](https://user-images.githubusercontent.com/8778433/37442793-3ccb4954-2843-11e8-803e-8e7ac5e86063.png)
![2018-03-15_112215](https://user-images.githubusercontent.com/8778433/37442794-3d106bba-2843-11e8-8e64-1d1f872fc932.png)

ubuntu enviroment

**Comment by yongfengnice** [comment]
make: *** No rule to make target 'libavfilter/vf_gltransition.c', needed by 'libavfilter/vf_gltransition.o'.  Stop

**Comment by yongfengnice** [comment]
I have compiled it successfully,but cannot use it.
the problem is below:


when I input: 
 ./ffmpeg -i out4.mp4 -i out5.mp4 -filter_complex gltransition -y out.mp4

it output errors:  
[AVFilterGraph @ 0x3509500] Error initializing filter 'gltransition'
Error initializing complex filters.
Operation not permitted

can you tell me how to solve?thanks

**Comment by transitive-bullshit** [comment]
it's an issue your your server's opengl setup. if it's headless, you need to run xvfb.

check this [thread](https://github.com/transitive-bullshit/ffmpeg-gl-transition/issues/5) for details.

**Comment by yongfengnice** [comment]
But I want to migrate to android platform.Do you think it can be used in android platform? can you help compile a lib and use it in android?thanks!!!

**Comment by FaithyQi** [comment]
I also want to use this lib in my Android project , but it doesn't build success. The problem is I can not build libglvnd with ffmpeg in NDK environment , can you help me ?  thanks!

**Comment by transitive-bullshit** [comment]
@FaithyQi make sure you can build normal ffmpeg (following the official ffmpeg Android docs) without this extension before trying to build with the extension added.

If that works for you and this extension doesn't work when applied, then please open a new issue with more details so we can investigate.

Thanks!

**Comment by lasnow** [comment]
Hi! I join this thread to see if we can make this work for Android. 
I was able to build [ffmpeg](https://github.com/IljaKosynkin/FFmpeg-Development-Kit) and ran it in 3 different Android devices. 

Now I am trying to put ffmpeg-gl-transition on top of my ffmpeg but the `vf-gltransition.c` file gives me compilation errors, there are some dependencies that don't exist or I don't know how to add them in the Android NDK compiler. 
`libavfilter/vf_gltransition.c:18:11: fatal error: 'GL/glew.h' file not found`
I [managed to solve](https://stackoverflow.com/questions/10793910/gl-h-glext-h-not-found) this by changing the import with `<GLES2/gl2.h>` 

But then I got this error: 
`libavfilter/vf_gltransition.c:25:11: fatal error: 'GLFW/glfw3.h' file not found`
So far, I haven't found any solution for that... Did anyone progress building ffmpeg-gl-transition for Android? 

Thanks
